src/lib/project-saver-hoc.jsx

```diff
@@ -143,6 +173,7 @@ const ProjectSaverHOC = function (WrappedComponent) {
          */
         storeProject (projectId, requestParams) {
             requestParams = requestParams || {};
+            this.clearAutoSaveTimeout();
```

Comment:
Hmm... do we want to clear it whenever we try to save or whenever actually accomplish saving? Maybe we just need this in `componentDidUpdate`?
```jsx
            if (this.props.projectChanged && !prevProps.projectChanged) {
                this.scheduleAutoSave();
            }
            if (!this.props.projectChanged && prevProps.projectChanged) {
                this.clearAutoSaveTimeout();
            }
```

Reply:
This is a compelling idea, and an elegant solution. But I think we do want to clear it whenever we try to save. I think an attempt to save which fails should leave us in a consistent state. This way, whether we're saving because we've received a timeout callback (which doesn't need clearing from `window`, as it is completed), or we are saving for some other reason (i.e., a manual save), we don't have the possibility of an additional attempt seconds later hanging around.

I think this sets us up well for save retrying.